application/src/main/java/org/opentripplanner/graph_builder/module/osm/VertexGenerator.java

```diff
@@ -95,32 +135,94 @@ IntersectionVertex getVertexForOsmNode(OsmNode node, OsmEntity way) {
         }
       }
 
-      if (node.isBarrier()) {
-        BarrierVertex bv = vertexFactory.barrier(nid, coordinate);
-        bv.setBarrierPermissions(node.overridePermissions(BarrierVertex.defaultBarrierPermissions));
-        iv = bv;
-      }
-
       if (includeOsmSubwayEntrances && node.isSubwayEntrance()) {
         String ref = node.getTag("ref");
-        iv = vertexFactory.stationEntrance(nid, coordinate, ref, node.wheelchairAccessibility());
+        iv = vertexFactory.stationEntrance(
+          nid,
+          coordinate,
+          ref,
+          node.explicitWheelchairAccessibility()
+        );
+      }
+
+      if (iv == null && node.isBarrier()) {
+        iv = vertexFactory.barrier(nid, coordinate, node.explicitWheelchairAccessibility());
+      }
+
+      if (iv instanceof BarrierVertex bv) {
+        bv.setBarrierPermissions(node.overridePermissions(BarrierVertex.defaultBarrierPermissions));
+        if (
+          bv.wheelchairAccessibility() == Accessibility.NO_INFORMATION &&
+          !node.isWheelchairAccessible()
+        ) {
+          bv.setWheelchairAccessibility(Accessibility.NOT_POSSIBLE);
+        }
       }
 
       if (iv == null) {
-        iv = vertexFactory.osm(
-          coordinate,
-          node,
-          node.hasHighwayTrafficLight(),
-          node.hasCrossingTrafficLight()
-        );
+        iv = vertexFactory.osm(node);
+      }
+
+      if (isNodeOnLinearBarrier && iv instanceof OsmVertex ov) {
+        splitVerticesOnBarriers.putIfAbsent(node, new HashMap<>());
+        var vertices = splitVerticesOnBarriers.get(node);
+        vertices.put(null, ov);
+
+        // some barrier values, such as barrier=gate, which do not block traversal will return
+        // false for node.isBarrier(), therefore we need to check the explicit absence of the
+        // barrier tag as well for the issue.
+        if (
+          !node.isBarrier() &&
+          !node.hasTag("barrier") &&
```

Comment:
Please add a method on `OsmNode` for this specific case and move the documentation to its Javadoc instead.